spec/forms/providers/means/identify_income_form_spec.rb

```diff
@@ -0,0 +1,299 @@
+require "rails_helper"
+
+# rubocop:disable Rails/SaveBang
+RSpec.describe Providers::Means::IdentifyIncomeForm do
+  describe "validations" do
+    context "when legal_aid_application_id is missing" do
+      it "is invalid" do
+        params = { "income_types" => [""] }
+
+        form = described_class.new(params)
+
+        expect(form).to be_invalid
+        expect(form.errors).to be_added(:legal_aid_application_id, :blank)
+      end
+    end
+
+    context "when neither a income type or none are selected" do
+      it "is invalid" do
+        params = { "legal_aid_application_id" => "test-id", "income_types" => [""] }
+
+        form = described_class.new(params)
+
+        expect(form).to be_invalid
+        expect(form.errors).to be_added(:income_types, :blank)
+      end
+    end
+
+    context "when at least one income type is selected, but an amount is missing" do
+      it "is invalid" do
+        benefits = create(:transaction_type, :benefits)
+        pension = create(:transaction_type, :pension)
+        params = {
+          "legal_aid_application_id" => "test-id",
+          "income_types" => ["", benefits.name, pension.name],
+          "benefits_amount" => "",
+          "benefits_frequency" => "weekly",
+          "pension_amount" => "100",
+          "pension_frequency" => "monthly",
+        }
```

Comment:
I constructed params like this to replicate how they might be submitted from the form in a real browser.